### ISSUE TITLE
Tagging a flow style map behaves differently from tagging a block style map

### ISSUE THREAD

**Issue by mononym** [yakaz/yamerl#34]
This likely happens for things like lists as well, but I didn't check those. But onto the issue.

I found this while implementing a keyword list tag for Elixir so that specific fields, rather than all map fields as is the only current option in `YamlElixir`, could be changed to keyword lists. So I implemented a tag and a corresponding module to process said tag and return the proper results.

When creating two different documents that look like so:
~~~~
foo:
  foo: bar
  bar: foo
~~~~
~~~~
foo: {foo: bar, bar: foo}
~~~~

The AST/results for the two documents look exactly the same. However if you try to tag the map under `foo` like so:
~~~~
foo: !<custom tag here>
  foo: bar
  bar: foo
~~~~
~~~~
foo: !<custom tag here> {foo: bar, bar: foo}
~~~~

The processing of these two documents take very different forms. In the first case the behaviour is as expected and effectively matches the flow in `yamerl_node_map.erl`. You get collection start/end tokens, and key/value tokens along with the unfinished node record to properly construct the data structure. In each of these cases the keys and values are tagged correctly so that the proper token can be constructed:`{:yamerl_scalar, 1, 1, {:yamerl_tag, 1, 1, {:non_specific, '?'}}, :flow, :plain, 'foo'}`

However when using the second, flow style, format the processing changes completely. Rather than receiving the above token to be constructed, instead we get this: `{:yamerl_scalar, 1, 44, {:yamerl_tag, 1, 6, 'tag:yaml_elixir,2019:keyword_list'}, :flow, :plain, 'foo'}`

Since that doesn't match the format expected by the `yamerl_node_str` module it won't build that token correctly. The Tag, rather than being applied to the structure as a whole, is being applied to the individual keys/values/everything within that flow style map. This means in order to support the flow style, the module implementing the custom tag now has to modify the token and then do its own iteration over the yamerl modules to see if they want to actually construct the token. Now I know there isn't exactly a lot of options to choose from when constructing the keys, but even there that seems like a bad blurring of responsibilities and leaking of internal logic.

What I would expect would be for the flow style map to behave exactly the same in processing as the block style map. With tags applied in the exact same way, and the processing done in the exact same order so that the same custom node module and logic would work for both styles.

**Comment by dumbbell** [comment]
Hi!

After one year, I confirm this bug... I don't know if I will have time to work on this anytime soon, as I don't work for Yakaz anymore. If you are still interested in that and are willing to contribute, that would be awesome!

**Comment by dumbbell** [comment]
In fact I took some time today and it was easier to find and fix than I thought. The fix will be included in yamerl 0.8.0, due today.
